contracts/Kernel.sol

```diff
@@ -59,27 +59,34 @@ contract Kernel is Factory {
         // TODO: we somehow need to execute this function
         // Let's see if we can decode the system call message at a higher level
         uint8 capType = uint8(msg.data[0]);
+        uint256 capIndex = 0;
         uint256 writeAddress = 0;
         uint256 writeValue = 0;
 
         for (uint256 i = 0; i < 32; i++) {
```

Comment:
Would move this to #checkWriteCapability

Reply:
Can you expand on what should be moved where?

Reply:
System call decoding has been moved into the relevant branches, although we need to do a little upfront obviously to choose the branch.